registry/storage/catalog.go

```diff
@@ -49,17 +48,22 @@ func (reg *registry) Repositories(ctx context.Context, repos []string, last stri
 
 		// if we've filled our array, no need to walk any further
 		if len(foundRepos) == len(repos) {
```

Comment:
This change looks fine but I am wondering why this check is done here rather than before line 45. The call returns immediately after adding to `foundRepos`, seems like adding to `foundRepos` should be followed up by an array size check.